### DISCUSSION TITLE
TypeScript definitions

### DISCUSSION THREAD

**Discussion by linusang** [jeremyckahn/shifty#132] [Ideas]
Any plans for TypeScript definitions?

**Comment by jeremyckahn** [comment]
Hi @linusang, thanks for bringing this up! I think TypeScript definitions would be a great addition to Shifty. Ideally Shifty would be authored in Typescript. A full TypeScript conversion could be interesting to explore for a future major version release. In the meantime, a complimentary definitions file would be a big step towards that.

To answer your specific question, I don't have any plans to write TypeScript definitions myself, as I'm not using Shifty in any TypeScript projects. However, I'd be happy to work with you or anyone else on a PR to add them!
